### PR TITLE
webui: a few tests fixes and code improvements 

### PR DESCRIPTION
**Pull request by KKoukiou**
Split out from https://github.com/rhinstaller/anaconda/pull/4922

### CODE REVIEW

File: ui/webui/src/components/storage/StorageConfiguration.jsx [hunk 1]
```diff
@@ -253,15 +251,49 @@ const GuidedPartitioning = ({ deviceData, selectedDisks, idPrefix, scenarios, st
...
+    }, []);
+
+    useEffect(() => {
+        const hasPartitions = selectedDisks.some(device => deviceData[device]?.children.v.some(child => deviceData[child]?.type.v === "partition"));
+
+        setHasPartitions(hasPartitions);
+    }, [selectedDisks, deviceData]);
 
     useEffect(() => {
         const updateScenarioState = async (scenarios) => {
```
Comment (rvykydal) [inline ui/webui/src/components/storage/StorageConfiguration.jsx:287]: Maybe all the hook content can become sync now?

**Review by VladimirSlavik — APPROVED** [review]
Looks good to me. Thank you!

File: ui/webui/src/components/storage/StorageConfiguration.jsx (hunk 1 above)
Comment (KKoukiou) [reply ui/webui/src/components/storage/StorageConfiguration.jsx:287]: Yes you 're right! thanks for noticing!

**Review by rvykydal — APPROVED** [review]
Looks good to me.

**Pull request merged**